### PR TITLE
Add a Question issue template

### PR DESCRIPTION
**Pull request by StephanTLavavej**
We have a `question` label for issues, but we currently don't encourage questions to be asked, nor do we provide guidance about what kinds of questions are on-topic.

Preview: https://github.com/StephanTLavavej/STL/issues/new/choose

### CODE REVIEW

**Comment by cbezault** [comment]
I think most users are going to just leave most of this template untouched when they submit a question. It isn't clear what sections the user is meant to fill out/modify and what sections are just there for their benefit.

**Comment by StephanTLavavej** [comment]
The template just provides guidance - users are expected to read it, delete it, and type their question. Should I add a sentence explicitly saying so?

**Comment by cbezault** [comment]
I understand the intent, but that would probably be helpful for others.

File: .github/ISSUE_TEMPLATE/question.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+* Would the maintainers be interested in specific enhancements?
+
+Off-Topic Examples
+------------------
+* Questions about non-STL components, such as the compiler, the CRT,
+the Windows API, the Visual Studio IDE, etc.
+* Questions about how to use the STL or C++ in general. For such questions,
+consider https://stackoverflow.com or https://www.reddit.com/r/cpp_questions/ .
+* Questions about whether you've encountered a bug in the STL. Instead of this
+Question issue template, please use the Bug Report issue template, because
```
Comment (AdamBucior) [inline .github/ISSUE_TEMPLATE/question.md:30]: I would change "issue template" to just "template". "Question template" and "Bug Report template" sound better.

Comment (StephanTLavavej) [reply .github/ISSUE_TEMPLATE/question.md:30]: Great idea, done!

File: .github/ISSUE_TEMPLATE/question.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+replace it with your question. If you aren't sure whether a question is
+on-topic, just go ahead and ask! :-)
+
+On-Topic Examples
+-----------------
+* What is this code in the STL doing? You can link to the relevant code:
+https://help.github.com/en/github/managing-your-work-on-github/creating-a-permanent-link-to-a-code-snippet
+* What are the preferred conventions for writing something?
+* What are the maintainers planning to do in the future?
+* Would the maintainers be interested in specific enhancements?
```
Comment (CaseyCarter) [inline .github/ISSUE_TEMPLATE/question.md:23]: I would like robot arms with laser cannon.

File: .github/ISSUE_TEMPLATE/question.md [hunk 3]
```diff
@@ -9,8 +9,10 @@ assignees: ''
 
 Instructions
 ============
-Here, you can ask a question about this STL implementation,
-and a maintainer will answer.
+Here, you can ask a question about this STL implementation, and a maintainer
+will answer. Please read the examples below, then delete all of this text and
+replace it with your question. If you aren't sure whether a question is
+on-topic, just go ahead and ask! :-)
```
Comment (SuperWig) [inline .github/ISSUE_TEMPLATE/question.md:15]: Probably not an issue but there's perhaps a slight ambiguity as to whether this means ask if the question is on topic or ask the question they were going to ask anyway.

Comment (StephanTLavavej) [reply .github/ISSUE_TEMPLATE/question.md]: Good catch, fixed!

**Comment by CaseyCarter** [comment]
Thanks for your questionable contribution! 😼

**Pull request merged**